src/librustc/middle/cstore.rs

```diff
@@ -268,6 +286,9 @@ pub trait CrateStore {
                                  reachable: &NodeSet)
                                  -> EncodedMetadata;
     fn metadata_encoding_version(&self) -> &[u8];
+
+    // access to the metadata loader
+    fn metadata_loader(&self) -> &MetadataLoader;
```

Comment:
Can you move this next to `crate_data_as_rc_any` (at the top of the trait)? They're both kind of special.